src/main/java/io/cdap/plugin/gcp/common/GCPUtils.java

```diff
@@ -310,4 +314,54 @@ public static String formatAsFQNComponent(String component) {
       return component;
     }
   }
+
+  /**
+   * Get a ProgramFailureException with the given error information.
+   *
+   * @param errorReason The reason for the error.
+   * @param errorMessage The error message.
+   * @param errorType The type of error.
+   * @return A ProgramFailureException with the given error information.
+   */
+  public static ProgramFailureException getProgramFailureException(Exception cause,
+      String errorReason, String errorMessage, ErrorType errorType, boolean dependency) {
+    ProgramFailureException.Builder builder = new ProgramFailureException.Builder();
+
+    if (cause != null) {
+      builder = builder.withCause(cause);
+    }
+
+    return builder
+        .withErrorCategory(new ErrorCategory(ErrorCategoryEnum.PLUGIN))
+        .withErrorReason(errorReason)
+        .withErrorMessage(errorMessage)
+        .withErrorType(errorType)
+        .withDependency(dependency)
+        .build();
+  }
+
+  /**
+   * Get a ProgramFailureException with the given error
+   * information from {@link GoogleJsonResponseException}.
+   *
+   * @param e The GoogleJsonResponseException.
+   * @return A ProgramFailureException with the given error information.
+   */
+  public static ProgramFailureException getProgramFailureException(GoogleJsonResponseException e) {
+    String errorReason = String.format("Request failed with error code: %s, status message: %s",
```

Comment:
This is not too actionable. Do we plan to improve it? The exception provides enough information to separate into network issues / auth issues / etc.

Reply:
Created new [ErrorUtils](https://github.com/cdapio/cdap/blob/fded74b9c1d470570b6ef7dbc71d9119e85e4062/cdap-api-common/src/main/java/io/cdap/cdap/api/exception/ErrorUtils.java#L89) class to have a mapping of known status codes.